standalone-metastore/metastore-rest-catalog/src/main/java/org/apache/iceberg/rest/HMSCatalogAdapter.java

```diff
@@ -0,0 +1,758 @@
+import org.apache.iceberg.rest.responses.ListNamespacesResponse;
+import org.apache.iceberg.rest.responses.ListTablesResponse;
+import org.apache.iceberg.rest.responses.LoadTableResponse;
+import org.apache.iceberg.rest.responses.LoadViewResponse;
+import org.apache.iceberg.rest.responses.OAuthTokenResponse;
+import org.apache.iceberg.rest.responses.UpdateNamespacePropertiesResponse;
+import org.apache.iceberg.util.Pair;
+import org.apache.iceberg.util.PropertyUtil;
+
+/**
+ * Original @ https://github.com/apache/iceberg/blob/1.6.x/core/src/test/java/org/apache/iceberg/rest/RESTCatalogAdapter.java
+ * Adaptor class to translate REST requests into {@link Catalog} API calls.
+ */
+public class HMSCatalogAdapter implements RESTClient {
+  /**  The metric names prefix. */
+  static final String HMS_METRIC_PREFIX = "hmscatalog.";
+  private static final Splitter SLASH = Splitter.on('/');
+
+  private static final Map<Class<? extends Exception>, Integer> EXCEPTION_ERROR_CODES =
+      ImmutableMap.<Class<? extends Exception>, Integer>builder()
+          .put(NamespaceNotSupported.class, 400)
+          .put(IllegalArgumentException.class, 400)
+          .put(ValidationException.class, 400)
+          .put(NamespaceNotEmptyException.class, 400)
+          .put(NotAuthorizedException.class, 401)
+          .put(ForbiddenException.class, 403)
+          .put(NoSuchNamespaceException.class, 404)
+          .put(NoSuchTableException.class, 404)
+          .put(NoSuchIcebergTableException.class, 404)
+          .put(UnsupportedOperationException.class, 406)
+          .put(AlreadyExistsException.class, 409)
+          .put(CommitFailedException.class, 409)
+          .put(UnprocessableEntityException.class, 422)
+          .put(CommitStateUnknownException.class, 500)
+          .buildOrThrow();
+
+  private static final String URN_OAUTH_TOKEN_EXCHANGE = "urn:ietf:params:oauth:grant-type:token-exchange";
+  private static final String URN_OAUTH_ACCESS_TOKEN = "urn:ietf:params:oauth:token-type:access_token";
+  private static final String GRANT_TYPE = "grant_type";
+  private static final String CLIENT_CREDENTIALS = "client_credentials";
+  private static final String BEARER = "Bearer";
+  private static final String CLIENT_ID = "client_id";
+  private static final String ACTOR_TOKEN = "actor_token";
+  private static final String SUBJECT_TOKEN = "subject_token";
+
+  private final Catalog catalog;
+  private final SupportsNamespaces asNamespaceCatalog;
+  private final ViewCatalog asViewCatalog;
+
+
+  public HMSCatalogAdapter(Catalog catalog) {
+    this.catalog = catalog;
+    this.asNamespaceCatalog =
+        catalog instanceof SupportsNamespaces ? (SupportsNamespaces) catalog : null;
+    this.asViewCatalog = catalog instanceof ViewCatalog ? (ViewCatalog) catalog : null;
+  }
+
+  enum HTTPMethod {
+    GET,
+    HEAD,
+    POST,
+    DELETE
+  }
+
+  enum Route {
+    TOKENS(HTTPMethod.POST, "v1/oauth/tokens",
```

Comment:
isn't that deprecated, see https://github.com/apache/iceberg/pull/10603/files#diff-02549ca620d020dc9ead80088cc14e311e12a69651fa8d394cd41a4308debb2e